sim/midas/src/main/cc/simif_token_hashers.cc

```diff
@@ -0,0 +1,222 @@
+#include "simif_token_hashers.h"
+#include "simif.h"
+
+#include <iostream>
+#include <sstream>
+#include <fstream>
+
+/**
+ * The constructor for simif_token_hashers_t.
+ *
+ * @param [in] p A pointer to the parent instance of simif_t
+ */
+simif_token_hashers_t::simif_token_hashers_t(simif_t *p) : parent(p) {
+
+#ifdef TOKENHASHMASTER_0_PRESENT
```

Comment:
In the spirit of the widget-like changes, consider adding this to `constructors.h` with an appropriate `add_token_hasher` call. Let's make that the only point where we access these macros.